xarray/coding/times.py

```diff
@@ -409,10 +449,16 @@ def decode_cf_datetime(
             dates = _decode_datetime_with_cftime(
                 flat_num_dates.astype(float), units, calendar
             )
+            # retrieve cftype
+            cftype = type(dates[np.nanargmin(num_dates)])
```

Comment:
I see that `dates[np.nanargmin(num_dates)]` is computed twice, maybe extract it into a variable once.

Also, is `type(dates[0])` also enough here (I'm not too familiar with cftime...)?

Reply:
> I see that `dates[np.nanargmin(num_dates)]` is computed twice, maybe extract it into a variable once.

Nice! I'll do so.

> Also, is `type(dates[0])` also enough here (I'm not too familiar with cftime...)?

Should work, too! Thanks!

Reply:
Used a combination of the above.